apps/CMakeLists.txt

```diff
@@ -134,18 +134,13 @@ if (BUILD_APPS)
 
     # Now external libs
     if (GDAL_USE_GEOS)
-      target_include_directories(${UTILCMD} PRIVATE ${GEOS_INCLUDE_DIRS})
+      target_include_directories(${UTILCMD} PRIVATE $<TARGET_PROPERTY:GEOS::GEOS,INTERFACE_INCLUDE_DIRECTORIES>)
```

Comment:
This is the spot which I don't like. 
- If the GEOS include directory would be really needed for all users fo lib gdal, it should already come as a public link library with lib gdal.
- If the GEOS include directory is needed for all apps here, I think it should be `target_link_libraries(${UTILCMD} PRIVATE GEOS::GEOS)`.
- If it is just `ogrlineref.cpp`, `target_link_libraries(...) should be called directly for the app which uses this source file.
- If it is just `ogrlineref.cpp`, it might be entirely obsolete after reviewing that file. Does the file use more than the GEOS version macros? Does GDAL still support older versions? (I really don't know.)

Reply:
GEOS 3.2 is really antiquated for GDAL master. I believe the HAVE_GEOS_PROJECT in this file could just be replaced by a simple HAVE_GEOS that we could pass with a target_compile_options, and that would remove the need for including ogr_geos.h.  It seems this is the only utility needed explicit use of GEOS